data-generation/src/main/scala/thesis/SnapshotDelta.scala

```diff
@@ -46,19 +46,26 @@ class SnapshotDelta(val graphs: MutableList[Snapshot],
 
   override def snapshotAtTime(instant: Instant): AttributeGraph = {
 
-    val closestGraph = graphs.reduce(returnClosestGraph(instant))
+    //Closest graph should be based on number of logs, not time.
+    // But for it to be feasible each graph should have a sequence ID as well
+    // This is possible, I think, but not implemented
+    val closestGraph = graphs.reduceLeft((snap1, snap2) =>
+      if (snap2.instant > instant) { 
+        snap1
+      } else {
+        snap2
+      })
     logger.warn(s"Instant $instant, Closest :graph${closestGraph.instant}")
     if (closestGraph.instant == instant) {
       logger.warn("The queried graph is already materialized")
       closestGraph.graph
-    } else if (closestGraph.instant.isBefore(instant)) {
-      logger.warn("Closest graph is in the future. Need to apply logs forwards")
+    } else if (instant < closestGraph.instant) { // We are asking an instant before the initial materialized snapshot
+      logger.warn("Closest graph in the future, and there is no one in the past.")
```

Comment:
```suggestion
    } else if (instant < closestGraph.instant) {
      logger.warn("There exists no snapshot before the given instant.")
```
Kommentar og logging sier litt det samme - er det dumt om vi bare beholder én?

Reply:
fixed